extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,426 @@
+require(
+	['jquery', 'wikia.window', 'wikia.log', 'ext.wikia.design-system.templating',
+		'ext.wikia.design-system.loading-spinner',
+		'ext.wikia.design-system.on-site-notifications.text-formatter',
+		'ext.wikia.design-system.on-site-notifications.common'],
+	function ($, window, log, templating, Spinner, TextFormatter, common) {
+		'use strict';
+
+		/**
+		 * Gets Date from ISO string date
+		 * @param {string} date
+		 * @returns {Date} - date
+		 */
+		function convertToTimestamp(date) {
+			return new Date(date);
+		}
+
+		/**
+		 * Gets ISO string from Date
+		 * @param {Date} date
+		 * @returns {string} - ISO string
+		 */
+		function convertToIsoString(date) {
+			return date.toISOString();
+		}
+
+		function getSafely(obj, path) {
```

Comment:
This method should be extracted to separate file (object utils?) and unit tested